src/PatternkitDrupalTwigLib.php

```diff
@@ -195,14 +247,123 @@ protected function getRawMetadata() {
    * @return string
    *   The rendered pattern HTML.
    */
-  public function getRenderedPatternMarkup(PatternkitPattern $pattern,
-    PatternkitEditorConfig $config) {
+  public function getRenderedPatternMarkup(
+    PatternkitPattern $pattern,
+    PatternkitEditorConfig $config
+  ) {
     if (empty($pattern->filename) || empty($config->fields)) {
       return '';
     }
     $template = $pattern->filename;
     $variables = $config->fields;
-    return twig_render_template($template, $variables);
+
+    return $this->renderTwigTemplate($template, $variables);
+  }
+
+  /**
+   * Returns a singleton version of the twig template engine.
+   *
+   * @return Twig_Environment
+   *   Twig environment object.
+   *
+   * @throws \Twig_Error_Loader
+   *   Twig engine instance object.
+   */
+  public function getTwigInstance() {
+    static $twig_engine;
+
+    if (!is_object($twig_engine)) {
+      // Setup twig environment.
+      // @TODO: Properly libraryize this.
+      require_once DRUPAL_ROOT . '/sites/all/libraries/Twig/Autoloader.php';
+      Twig_Autoloader::register();
+
+      $loader = new Twig_Loader_Filesystem();
+
+      $metadata = $this->getRawMetadata();
+      foreach ($metadata as $module_name => $module) {
+        if (!empty($module->filename)) {
+          $templatesDirectory = DRUPAL_ROOT . DIRECTORY_SEPARATOR . dirname(
+              $module->filename
+            );
+
+          // Namespacing is not necessary here, as each "engine" is unique to
+          // the "library".
+          $loader->addPath($templatesDirectory);
```

Comment:
Haven't tested yet, but does this preclude support twig libraries that depend on namespaces?

Will update review comment if I get a chance to test it out before you get to this.

Reply:
no, you can add a namespace here. I just didn't because then I'd have to overload the FileLoader. I actually started to do that and then realized that I didn't need them.

Reply:
overload -> subclass